docs/adr/0011-operators.md

```diff
@@ -0,0 +1,145 @@
+# Managing operators
+
+- Status: accepted
+- Deciders: tumido, HumairAK, 4n4nd, anishasthana
+- Date: 2021-01-27
+
+Technical Story: [issue-1](https://github.com/operate-first/apps/issues/126), [pr-1](https://github.com/operate-first/apps/pull/131), [pr-2](https://github.com/operate-first/apps/pull/153)
+
+## Context and Problem Statement
+
+A `cluster-scope` application definition introduced by [ADR 0010][] is a bit vague for use cases like deploying operators. Operator Lifecycle Manager (OLM) defines a resource overlap where a single resource kind can be used in different contexts which may go against the distinction introduced in [ADR 0010][]. This document should provide guidance to how each of the resource kinds for OLM should be treated in each context and usage variant we aim to support.
+
+## Decision Drivers
+
+- The definition of a `cluster-scope` application at [ADR 0010][] does not provide enough guidance for OLM.
+- Various OLM resources can be used in a way which effects the whole cluster or in a way that limits the effects to a single namespace.
+- Various OLM resources have permission restrictions or behave as a singletons within a namespace.
+- Some operators are not deployed via OLM, yet they behave as operators across all namespaces. Placement of those manifests is also disputed.
+- How should we handle meta-operators which operate across all namespaces and deploy other operators that operate across all namespaces?
+
+## Considered Options
+
+1. Deploy all operator-related resources within `cluster-scope` application.
+2. Require all namespaced resources to be deployed within individual applications, outside of `cluster-scope`.
+3. Find balance between 1. and 2. Deploy resources which have effect on a cluster level within `cluster-scope` app (also the namespace-scoped ones). The rest is deployed in individual applications.
+
+## Decision Outcome
+
+Chosen option: _"3. Find balance between 1. and 2."_, because it is desired to keep track of all resources which may effect other users and other namespaces i.e. operators provisioned to manage all cluster namespaces.
+
+Proposed decision defines strict rules for given resources. These rules are subject to future updates if needed, but the repositories and applications are required to consistently enforce them:
+
+### Operator Lifecycle Manager compatible operators
+
+Operator Lifecycle Manager defines following resource kinds:
+
+- Cluster service version
+- Catalog source
+- Subscription
+- Install plan
+- Operator groups
+
+These are the set of guidelines we want to enforce on them:
+
+#### Cluster service version
+
+This resource is not meant to be deployed manually and is expected to be provided from a _Catalog source_. If there's really a need to deploy a custom _Cluster service version_, the resource has to live within the `cluster-scope` application.
```

Comment:
Do we differentiate here between namespace/global operators? For instance, in the very unlikely scenario we need to do this, if it's global we store this in `openshift-operators` namespace yes? otherwise it goes in the appropriate user namespace under cluster-scope. Should we mention that here?

Reply:
I think we can and sometimes want different global operators to live in different namespaces. So we can for example enforce quotas. To answer your question, yes we differentiate between namespaces and global operators.

Sometimes even the cluster scoped operator has a namespace preference/requirement... It can be messy.

Reply:
So do we then, leave the judgement of where/how a global operator ought to be deployed on a case by case basis as it seems like there maybe a myriad of different reasons why we might want it to be in `openshift-operators` or in its own namespace?

Reply:
I'd say let's decide on case by case basis, but default to `openshift-operators` if we have no solid reason why it should live in a separate namespace. WDYT?

Reply:
Yeah I'm fine with this. I think we should mention  something like this somwehere then: 
> Deciding which namespace the OLM resources should reside for global operators will be done on a case-by-case basis. In general, the rule of thumb is to put them in `openshift-operators`, however we take note that there may be various exceptions to this rule (e.g. ClusterLogging, Openshift Metering, etc.).

Reply:
I like that. Applied. :+1:

Reply:
Btw, I'm gonna put this into the `Subscription` section rather than here, since the subscription resource defines the operator placement. Sounds good?

Reply:
yep that's fine

Reply:
+ I'll add to Subscription:

```
Deciding which namespace the OLM resources should reside for global operators will be done on a case-by-case basis. In general, the rule of thumb is to put them in `openshift-operators`, however we take note that there may be various exceptions to this rule (e.g. ClusterLogging, OpenShift Metering, etc.).

The `Subscription` resource manifest is expected to be deployed to the same namespace where the corresponding `OperatorGroup` is located.
```

And to `OperatorGroup`:
```
The `OperatorGroup` resource manifest is expected to be deployed to the same namespace where the corresponding `Subscription` is located.
```

So it's really clear

Reply:
perfect!